core/github.go

```diff
@@ -24,6 +26,10 @@ func CloneGithubRepository(cloneConfig *CloneConfiguration) (*git.Repository, st
 		ReferenceName: plumbing.ReferenceName(fmt.Sprintf("refs/heads/%s", *cloneConfig.Branch)),
 		SingleBranch:  true,
 		Tags:          git.NoTags,
+		Auth: &http.BasicAuth{
```

Comment:
I'll work on using the builtin Auth, but I believe CheckGithubAPIToken is still getting called in session.go before we even get to this point in execution.

Reply:
After a good deal of looking, I'm not sure the auth can be easily pulled out from CloneGithubRepository to NewClient. Best i'm coming up with is adding that basicauth object as part of the session object and referencing it that way. Thoughts? Possible I'm misunderstanding or missing something though.

Reply:
Honestly, rather than overcomplicate this. Lets get it in, tech debt it, and then refactor if it becomes an issue. I want all auth together as it makes sense but its not a high I want to die on. I also don't want to solve problems we don't yet have. @funkwhatyouheard